plugins/inputs/cloudwatch/cloudwatch_test.go

```diff
@@ -6,48 +6,98 @@ import (
-	metric := &cloudwatch.Metric{
-		Namespace:  params.Namespace,
-		MetricName: aws.String("Latency"),
-		Dimensions: []*cloudwatch.Dimension{
+	return &cloudwatch.ListMetricsOutput{
+		Metrics: []*cloudwatch.Metric{
 			{
-				Name:  aws.String("LoadBalancerName"),
-				Value: aws.String("p-example"),
+				Namespace:  params.Namespace,
+				MetricName: aws.String("Latency"),
+				Dimensions: []*cloudwatch.Dimension{
+					{
+						Name:  aws.String("LoadBalancerName"),
+						Value: aws.String("p-example"),
+					},
+				},
 			},
 		},
-	}
-
-	result := &cloudwatch.ListMetricsOutput{
-		Metrics: []*cloudwatch.Metric{metric},
-	}
-	return result, nil
+	}, nil
 }
 
-func (m *mockGatherCloudWatchClient) GetMetricStatistics(params *cloudwatch.GetMetricStatisticsInput) (*cloudwatch.GetMetricStatisticsOutput, error) {
-	dataPoint := &cloudwatch.Datapoint{
-		Timestamp:   params.EndTime,
-		Minimum:     aws.Float64(0.1),
-		Maximum:     aws.Float64(0.3),
-		Average:     aws.Float64(0.2),
-		Sum:         aws.Float64(123),
-		SampleCount: aws.Float64(100),
-		Unit:        aws.String("Seconds"),
-	}
-	result := &cloudwatch.GetMetricStatisticsOutput{
-		Label:      aws.String("Latency"),
-		Datapoints: []*cloudwatch.Datapoint{dataPoint},
-	}
-	return result, nil
+func (m *mockGatherCloudWatchClient) GetMetricData(params *cloudwatch.GetMetricDataInput) (*cloudwatch.GetMetricDataOutput, error) {
+	return &cloudwatch.GetMetricDataOutput{
+		MetricDataResults: []*cloudwatch.MetricDataResult{
+			{
+				Id:         aws.String("minimum_0"),
+				Label:      aws.String("latency_minimum"),
+				StatusCode: aws.String("completed"),
+				Timestamps: []*time.Time{
+					params.EndTime,
+				},
+				Values: []*float64{
+					aws.Float64(0.1),
+				},
+			},
+			{
+				Id:         aws.String("maximum_0"),
+				Label:      aws.String("latency_maximum"),
+				StatusCode: aws.String("completed"),
+				Timestamps: []*time.Time{
+					params.EndTime,
+				},
+				Values: []*float64{
+					aws.Float64(0.3),
+				},
+			},
+			{
+				Id:         aws.String("average_0"),
+				Label:      aws.String("latency_average"),
+				StatusCode: aws.String("completed"),
+				Timestamps: []*time.Time{
+					params.EndTime,
+				},
+				Values: []*float64{
+					aws.Float64(0.2),
+				},
+			},
+			{
+				Id:         aws.String("sum_0"),
+				Label:      aws.String("latency_sum"),
+				StatusCode: aws.String("completed"),
+				Timestamps: []*time.Time{
+					params.EndTime,
+				},
+				Values: []*float64{
+					aws.Float64(123),
+				},
+			},
+			{
+				Id:         aws.String("sample_count_0"),
+				Label:      aws.String("latency_sample_count"),
+				StatusCode: aws.String("completed"),
+				Timestamps: []*time.Time{
+					params.EndTime,
+				},
+				Values: []*float64{
+					aws.Float64(100),
+				},
+			},
+		},
+	}, nil
 }
 
+func TestSnakeCase(t *testing.T) {
+	assert.Equal(t, "cluster_name", snakeCase("Cluster Name"))
+	assert.Equal(t, "broker_id", snakeCase("Broker ID"))
+}
```

Comment:
nit ... add a space between these functions.

Reply:
Can't believe I missed that